### PR TITLE
Upgrade rails 5.1.6.2 to 5.2.4.3

### PR DESCRIPTION
**Pull request by kayison**
## Description

Upgrade rails to fix security vulnerabilities in earlier versions. 

There are some Rails 6 deprecation errors that have been silenced temporarily to enable us to upgrade to fix the security vulnerabilities. 

Some rubocop changes as part of the upgrade. Add ignores to existing migration files so we don't modify them but leave the rule open for new ones.

## CCs

If app migration related
@zendesk/volunteer

Anyone specific?

## Risks (if any)
* Low: test to make sure rails upgrade is ok

### CODE REVIEW

File: db/migrate/20161211042811_add_timestamps_to_offices.rb [hunk 1]
```diff
@@ -1,7 +1,9 @@
+# rubocop:disable Rails/BulkChangeTable
 class AddTimestampsToOffices < ActiveRecord::Migration[4.2]
   def change
     add_column :offices, :created_at, :datetime
```
Comment (samzx) [inline db/migrate/20161211042811_add_timestamps_to_offices.rb:4]: not too familiar with rails, but it seems like these can be combined into one somehow?

**Review by samzx — APPROVED** [review]
we can have a look at combining the queries in another pr if possible

File: db/migrate/20161211042811_add_timestamps_to_offices.rb (hunk 1 above)
Comment (kayison) [reply db/migrate/20161211042811_add_timestamps_to_offices.rb:4]: Do you mean adding something to the `.rubocop.yml`?

Comment (samzx) [reply db/migrate/20161211042811_add_timestamps_to_offices.rb:4]: never mind, I think this is fine, unless we can ignore the directory somehow

Comment (kayison) [reply db/migrate/20161211042811_add_timestamps_to_offices.rb:4]: I'm ignoring existing migrations so we don't modify them and keeping the rule open for any new migrations

**Pull request merged**